vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/frequency_capping.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/frequency_capping/frequency_capping.h"
+#include "bat/ads/ad_info.h"
+#include "bat/ads/internal/client.h"
+#include "bat/ads/internal/time.h"
+
+namespace ads {
+
+bool FrequencyCapping::DoesHistoryRespectCapForRollingTimeConstraint(
+    const std::deque<uint64_t> history,
+    const uint64_t time_constraint_in_seconds,
+    const uint64_t cap) const {
+  uint64_t count = 0;
+
+  auto now_in_seconds = Time::NowInSeconds();
+
+  for (const auto& timestamp_in_seconds : history) {
+    if (now_in_seconds - timestamp_in_seconds < time_constraint_in_seconds) {
+      count++;
+    }
+  }
+
+  if (count < cap) {
+    return true;
+  }
+
+  return false;
+}
+
+std::deque<uint64_t> FrequencyCapping::GetCreativeSetHistoryForUuid(
+    const std::string& uuid) const {
+  std::deque<uint64_t> history;
+
+  auto creative_set_history = client_->GetCreativeSetHistory();
+  if (creative_set_history.find(uuid) != creative_set_history.end()) {
+     history = creative_set_history.at(uuid);
+  }
+
+  return history;
+}
+
+std::deque<uint64_t> FrequencyCapping::GetAdsHistory() const {
```

Comment:
 GetAdsHistory but should be GetAdsShownHistory? and now 'History' is timestamps, not AdHistoryDetail's... confusing the client_->GetAdsShownHistory

Reply:
Agreed `This should be `GetAdsShowwHistory`

Reply:
Done